domain-models-runtime/src/main/resources/templates/db_scripts/optimistic_locking.ftl

```diff
@@ -0,0 +1,50 @@
+-- auto update optimistic locking version
+-- ERROR code 23F09: 23 for class 23 — Integrity Constraint Violation, F for FOLIO, 09 for 409 HTTP status code
+
+<#if table.withOptimisticLocking??>
+  <#assign ol_version = "_version">
+  CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.${table.tableName}_set_ol_version()
+  RETURNS trigger AS $$
+  BEGIN
+    CASE TG_OP
+      WHEN 'INSERT' THEN
+        <#if table.withOptimisticLocking.name() != "OFF">
+          NEW.jsonb = jsonb_set(NEW.jsonb, '{${ol_version}}', to_jsonb(1));
+        </#if>
+      WHEN 'UPDATE' THEN
+        <#if table.withOptimisticLocking.name() == "OFF">
+          NEW.jsonb = NEW.jsonb - '${ol_version}';
```

Comment:
Removing the version does not match the documentation.
If optimistic locking is off the trigger should be removed if it exists.

Reply:
> Removing the version does not match the documentation.
> If optimistic locking is off the trigger should be removed if it exists.

Did not know we should remove the trigger when config is changed to `off`. Made the change as suggested now. Technically speaking, since this `off` is the same as the config is not defined, maybe we should simplify it by removing this `off` option.